lua/rustowl/lsp.lua

```diff
@@ -0,0 +1,70 @@
+local M = {}
+
+---@param filter? vim.lsp.get_clients.Filter
+---@return vim.lsp.Client[]
+M.get_rustowl_clients = function(filter)
+  filter = vim.tbl_deep_extend('force', filter or {}, {
+    name = 'rustowl',
+  })
+  return vim.lsp.get_clients(filter)
+end
+
+---Start / attach the LSP client
+---@return integer|nil client_id The LSP client ID
+M.start = function()
+  local config = require('rustowl.config')
+  local client_config = vim.tbl_deep_extend('force', {}, config.client)
+  client_config.root_dir = config.client.root_dir()
+  return vim.lsp.start(client_config)
+end
+
+---Compatibility for a breaking change in Nvim 0.11
+---@param client vim.lsp.Client
+---@return boolean
+local function client_is_stopped(client)
+  local info = debug.getinfo(client.is_stopped, 'u')
+  if info.nparams > 0 then
+    ---@diagnostic disable-next-line: param-type-mismatch
+    return client:is_stopped()
+  else
+    ---@diagnostic disable-next-line: missing-parameter
+    return client.is_stopped()
+  end
+end
+
+M.stop = function()
+  local clients = M.get_rustowl_clients()
+  vim.lsp.stop_client(clients, true)
+  local timer, _, _ = vim.uv.new_timer()
```

Comment:
This function has multiple missing `nil` checks on `timer`. I'm not sure what the best way is, but one simple way is this
```suggestion
  local t, err, _ = vim.uv.new_timer()
  local timer = assert(t, err)
```
Maybe not optimal but slightly better